src/medley/core.cljc

```diff
@@ -45,11 +45,13 @@
   ([m k v]
    (if (nil? v) m (assoc m k v)))
   ([m k v & kvs]
-   (loop [m (assoc-some-transient! (transient m) k v)
+   (loop [acc (assoc-some-transient! (transient (or m {})) k v)
           kvs kvs]
-     (if (next kvs)
-       (recur (assoc-some-transient! m (first kvs) (second kvs)) (nnext kvs))
-       (persistent! m)))))
+     (cond
+       (next kvs) (recur (assoc-some-transient! acc (first kvs) (second kvs))
+                         (nnext kvs))
+       (zero? (count acc)) m ; see CLJ-1872
+       :else (persistent! acc)))))
```

Comment:
What about using two `if` forms instead? I think makes it clearer that `(next kvs)` is designed to short circuit `(zero? (count acc))`, and it's no more verbose.

```clojure
     (if (next kvs)
       (recur (assoc-some-transient! acc (first kvs) (second kvs)) (nnext kvs))
       (if (zero? (count acc))
         m
         (persistent! acc)))
```

Reply:
Sounds good to me, and better matches other fns in the namespace.